adapter/oam.go

```diff
@@ -216,9 +223,78 @@ func CreateComponents(scfg StaticCompConfig) error {
 			return ErrCreatingComponents(err)
 		}
 	}
+	err = copyCoreComponentsToNewVersion(filepath.Join(scfg.Path, dirName), filepath.Join(scfg.Path, scfg.DirName), scfg.DirName)
+	if err != nil {
+		return ErrCreatingComponents(err)
+	}
 	return nil
 }
 
+//Currently versioning of Meshery's core components from a given service mesh adapter is corresponding
+//to the service mesh version itself. So everytime components are generated for a new service mesh version,
+// existing core components from previously latest version are to be copied in this new version.
+//Further we can change the schema of copied core components in latest version, or leave it as it is.
+func copyCoreComponentsToNewVersion(fromDir string, toDir string, newVersion string) error {
+	files, err := ioutil.ReadDir(fromDir)
+	if err != nil {
+		return err
+	}
+	for _, f := range files {
+		//core definition file or core schema file
+		if !strings.Contains(strings.TrimSuffix(f.Name(), ".json"), ".") || !strings.Contains(strings.TrimSuffix(f.Name(), ".meshery.layer5io.schema.json"), ".") {
+			fsource, err := os.Open(filepath.Join(fromDir, f.Name()))
+			if err != nil {
+				return err
```

Comment:
No new or existing error code to use here?

Reply:
These are helper functions and these errors will be wrapped inside meshkit errors before being thrown.